experimental/devices/bitbang/spi.go

```diff
@@ -70,8 +70,11 @@ func (s *SPI) Connect(f physic.Frequency, mode spi.Mode, bits int) (spi.Conn, er
 	if f < 0 {
 		return nil, errors.New("bitbang-spi: invalid frequency")
 	}
-	if mode != spi.Mode3 {
-		return nil, fmt.Errorf("bitbang-spi: mode %v is not implemented", mode)
+	if mode&spi.HalfDuplex == spi.HalfDuplex {
+		return nil, errors.New("bitbang-spi: half-duplex mode not supported")
+	}
+	if mode&spi.LSBFirst == spi.LSBFirst {
+		return nil, errors.New("bitbang-spi: LSBFirst mode not supported")
 	}
```

Comment:
In practice, you'd want to check explicitly for 0 <= mode <= 3, so that if new flags are added, they are not silently ignored.

Reply:
Good point.